Documentation/compatibility/wpf-accessibility-improvements-48.md

```diff
@@ -0,0 +1,205 @@
+
+##### ControllerFor property support
+UIAutomation�s ControllerFor property returns an array of automation elements that are manipulated by the automation element that supports this property. This property is commonly used for Auto-suggest accessibility. ControllerFor is used when an automation element affects one or more segments of the application UI or the desktop. Otherwise, it is hard to associate the impact of the control operation with UI elements. This feature adds the ability for controls to provide a value for ControllerFor property.
+
+A new virtual method has been added to [AutomationPeer](xref:System.Windows.Automation.Peers.AutomationPeer):
+```csharp
+    virtual protected List<AutomationPeer> GetControlledPeersCore()
+```
+To provide a value for the ControllerFor property, simply override this method and return a list of [AutomationPeers](xref:System.Windows.Automation.Peers.AutomationPeer) for the controls being manipulated by this [AutomationPeer](xref:System.Windows.Automation.Peers.AutomationPeer):
+```csharp
+    
+public class AutoSuggestTextBox: TextBox
+    {
+        protected override AutomationPeer OnCreateAutomationPeer()
+        {
+            return new AutoSuggestTextBoxAutomationPeer(this);
+        }
+
+        public ListBox SuggestionListBox;
+    }
+
+    internal class AutoSuggestTextBoxAutomationPeer : TextBoxAutomationPeer
+    {
+        public AutoSuggestTextBoxAutomationPeer(AutoSuggestTextBox owner) : base(owner)
+        {        
+        }
+
+        protected override List<AutomationPeer> GetControlledPeersCore()
+        {
+            List<AutomationPeer> controlledPeers = new List<AutomationPeer>();
+            AutoSuggestTextBox owner = Owner as AutoSuggestTextBox;
+
+            controlledPeers.Add(UIElementAutomationPeer.CreatePeerForElement(owner.SuggestionListBox));
+            return controlledPeers;
+        }
+    }
+```
+
+##### Name Automation property is now correctly propagated for DatePicker and RibbonComboBox controls
+Fixed an accessibility problem where [DatePicker](xref:System.Windows.Controls.DatePicker) and [RibbonComboBox](xref:System.Windows.Controls.Ribbon.RibbonComboBox) wouldn't reflect the [AutomationProperties.Name](xref:System.Windows.Automation.AutomationProperties.Name) value assigned to it when read by a screen reader.
+
+##### Expanders in GroupItem headers are now announced by screen readers
+Fixed an accessibility problem where an [Expander](xref:System.Windows.Controls.Expander) in a re-templated [GroupItem's](xref:System.Windows.Controls.GroupItem) header was not being announced by screen readers.
+
+##### Improved focus move logic when WPF is hosted in Windows Forms
+Fixed an accessibility problem where focus loops inside a WPF [UserControl](xref:System.Windows.Controls.UserControl) instead of breaking out of it under some hosting scenarios
```

Comment:
```suggestion
The focus no longer loops inside a WPF <xref:System.Windows.Controls.UserControl> instead of breaking out of it under some hosting scenarios.
```